cmd/bench.go

```diff
@@ -316,6 +323,30 @@ func benchE2E(ctx context.Context, args []string, params benchmarkCommandParams,
 		break
 	}
 
+	// Busy loop until server has truly come online to recover the bound port.
+	// We do this with exponential backoff for wait times, since the server
+	// typically comes online very quickly.
+	baseDelay := time.Duration(10) * time.Microsecond
+	maxDelay := time.Duration(60) * time.Second
+	retries := 8 // Max of around 1 minute total wait time.
+	for i := 0; i < retries; i++ {
+		if len(rt.Addrs()) == 0 {
+			delay := util.DefaultBackoff(float64(baseDelay), float64(maxDelay), i)
+			time.Sleep(delay)
+			continue
+		}
+		// We have an address to parse the port from.
+		port, err = strconv.Atoi(strings.Split(rt.Addrs()[0], ":")[1])
+		if err != nil {
+			return err
+		}
+		break
+	}
+	// Check for port still being unbound after retry loop.
+	if len(rt.Addrs()) == 0 {
```

Comment:
Let's check if port is != 0 instead. It could be that this is _true_ when we evaluate this line; but was false all through the loop.

Reply:
That sounds like it'll get us closer to where we want to be. I've made that change, and we'll see what shakes out in CI.